### PR TITLE
Feature: mTLS

### PR DESCRIPTION
**Pull request by sgammon**
## Description
Introduce functionality to allow mTLS when given certificate assets and a private key.

Related issues:
- BLOOM-58 #comment Begin adding mTLS to Java client.
- BLOOM-59 #comment Java client functionality for mTLS.

Worklog:
- [x] Sync schema with `release/1.4.6`
- [x] Provide constructors/settings for client keys and certs
- [x] Setup `RPCClient` to work this way
- [x] Version bump -> `beta11`

## Environment
Language Version:
- [x] JDK7
- [x] JDK8
- [x] JDK9
- [x] Kotlin

## Services
- [x] Shop API
- [x] Telemetry API
- [x] Menu API
- [x] Checkin API
- [x] Devices API

### CODE REVIEW

File: src/main/kotlin/bloombox/client/services/shop/ShopClient.kt [hunk 1]
```diff
@@ -110,15 +116,14 @@ class ShopClient(override val host: String,
...
-        .intercept(interceptor)
-        .build()
+  override val channel: ManagedChannel = channelBuilder(
+        host = host,
+        port = port,
+        executor = executor,
+        clientAuth = clientAuth,
+        transportMode = transportMode,
+        clientCredentials = clientCredentials,
+        clientAuthorityRoots = clientAuthorityRoots).intercept(interceptor).build()
```
Comment (sgammon) [inline src/main/kotlin/bloombox/client/services/shop/ShopClient.kt:126]: new syntax for creating a client-side `ManagedChannel`, with support for crypto assets

File: src/main/kotlin/bloombox/client/services/telemetry/TelemetryClient.kt [hunk 2]
```diff
@@ -330,24 +332,18 @@ class TelemetryClient(override val host: String,
...
 
       // note when the pong came back
-      if (callback != null) {
-        val done = System.currentTimeMillis()
-        callback(done - start!!)
-      }
+      callback?.invoke(end - start)
     } catch (e: StatusRuntimeException) {
-
+      callback?.invoke(null)
```
Comment (sgammon) [inline src/main/kotlin/bloombox/client/services/telemetry/TelemetryClient.kt:346]: generic fixes for warnings reported by Codacy

File: src/main/kotlin/bloombox/client/test/ClientRPCTest.kt [hunk 3]
```diff
@@ -75,9 +74,17 @@ open class ClientRPCTest {
...
+
+  /**
+   * Provide a callback with a setup client, and then immediately
+   * close it afterwards.
+   */
+  fun withClient(settings: Bloombox.Settings? = null, block: (RPCClient) -> Unit) {
     var client: RPCClient? = null
     try {
-      client = RPCClient()
+      client = RPCClient(settings)
```
Comment (sgammon) [inline src/main/kotlin/bloombox/client/test/ClientRPCTest.kt:87]: made `settings` in `ClientRPCTest` overridable, moved it to the main library so we can access it in platform tests

File: src/main/kotlin/bloombox/client/services/menu/MenuClient.kt [hunk 4]
```diff
@@ -66,12 +72,21 @@ class MenuClient(override val host: String,
...
+       * API key header sentinel.
+       */
       val apiKeyHeader: io.grpc.Metadata.Key<String> = io.grpc.Metadata.Key.of(
             "x-api-key", io.grpc.Metadata.ASCII_STRING_MARSHALLER)
+
+      /**
+       * Menu fingerprint header sentinel.
+       */
+      val menuFingerprintHeader: io.grpc.Metadata.Key<String> = io.grpc.Metadata.Key.of(
+            "x-menu-fingerprint", io.grpc.Metadata.ASCII_STRING_MARSHALLER)
```
Comment (sgammon) [inline src/main/kotlin/bloombox/client/services/menu/MenuClient.kt:89]: we'll begin using menu fingerprints soon enough

**Review by sgammon — COMMENTED** [review]
most of the other stuff is just schema changes

File: src/main/java/io/bloombox/schema/partner/PartnerScope.java [hunk 5]
```diff
@@ -1,70 +1,78 @@
...
   DASHBOARD(2),
   /**
    * <pre>
-   * - Menus
+   * Authorizes access to read menu data for a given partner account.
    * </pre>
    *
    * <code>MENU_DATA_READ = 10;</code>
    */
   MENU_DATA_READ(10),
```
Comment (tyler-porras) [inline src/main/java/io/bloombox/schema/partner/PartnerScope.java:46]: I assume we jump from **2** to **10** to allow for other types of product accesses in the future

Comment (sgammon) [reply src/main/java/io/bloombox/schema/partner/PartnerScope.java:46]: exactly - in case we need some other core ones between 2 and 10.

**Review by tyler-porras — APPROVED** [review]
Only 80,114 insertions. Up those numbers kid ;)))))))))

**Pull request merged**